### PR TITLE
Speed up builds by having make run multiple simultaneous jobs.

### PR DESCRIPTION
**Pull request by ltratt**
This speeds up builds significantly. I know it's a rounding error compared to the time to run experiments, but when one has to rebuild to test something, saving a couple of hours definitely helps.

Should only be considered for merger after #182 has been merged.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -52,6 +52,20 @@ case `uname` in
 esac
 check_for ${GMAKE}
 
+# How many cores does this machine have (so we can pass a sensible number to
+# gmake -j)?
+num_cores=1
+case `uname` in
+    OpenBSD) num_cores=`sysctl hw.ncpu|cut -d "=" -f 2`;;
+    Linux) num_cores=`cat /proc/cpuinfo|grep cores|head -n 1|cut -d ":" -f 2|sed "s/ //g"`;;
```
Comment (vext01) [inline build.sh]: On linux, you can use `nproc`:

```
root@bencher3:~# nproc
4
```

File: build.sh [hunk 2]
```diff
@@ -52,6 +52,20 @@ case `uname` in
...
+esac
+# Keep num_cores to safe limits for pretty much any machine out there.
+if [ $num_cores -le 0 ]; then
```
Comment (vext01) [inline build.sh]: Does this case happen? Should we perhaps bomb out instead?

**Comment by vext01** [comment]
Looks OK bar minor comments.

File: build.sh (hunk 1 above)
Comment (ltratt) [reply build.sh]: Aha!

File: build.sh (hunk 2 above)
Comment (ltratt) [reply build.sh]: This is just paranoia on my part: since we don't _need_ to increase the number of jobs, 1 is always a safe default.

**Comment by ltratt** [comment]
If that latest commit works for you, I'll squash, and then we should be good for merging.

**Comment by vext01** [comment]
Looks good. I'll do a quick test on a bencher.

**Comment by ltratt** [comment]
OK.

**Comment by vext01** [comment]
OK, i see gcc being built on 4 busy cores.

Squash?

**Comment by ltratt** [comment]
OK, squashed.

**Pull request merged**